### ISSUE TITLE
Allow numeric literal to be safely coerced to concrete numeric types

### ISSUE THREAD

**Issue by adamperlin** [influxdata/platform#255]
From ifql created by [nathanielc](https://github.com/nathanielc) : influxdata/ifql#309  

From the Spec:

Numeric literals may be integers or floating point values.
Literals have arbitrary precision and will be coerced to a specific type when used.

The following coercion rules apply to numeric literals:

* an integer literal can be coerced to an "int", "uint", or "float" type,
* an float literal can be coerced to a "float" type,
* an error will occur if the coerced type cannot represent the literal value.

**Comment by aanthony1243** [comment]
@nathanielc   @adamperlin   I don't believe this is implemented, but we can prepare a test for it anyway since it's in the spec.
